### PR TITLE
Beta release testing

### PR DESCRIPTION
**Pull request by Kelbaso**
The Most Up To Date Testing For The Bata Release With Code Quality Improvements

### CODE REVIEW

File: src/backend/gamelauncher.cpp [hunk 1]
```diff
@@ -30,13 +30,17 @@ namespace launcher {
...
+            cout<<"Create Process Success"<<endl;
         }
-        cout<<"Create Process Success"<<endl;
+
+        // IF PROCCESS ID OR THREAD ID IS WANTED UNCOMMENT THE FOLLOWING LINES
 
         // Terminal will help us track the proccess and thread
-        cout<<"Process ID ->"<<processinfo.dwProcessId<<endl;
-        cout<<"Thread ID ->"<<processinfo.dwThreadId<<endl;
+        // cout<<"Process ID ->"<<processinfo.dwProcessId<<endl;
```
Comment (lochoa2) [inline src/backend/gamelauncher.cpp:42]: delete code you don't use

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ jobs:
       working-directory: ${{env.GITHUB_WORKSPACE}}
       run: VSTest.Console.exe ${{env.TEST_BIN_PATH}}/${{env.BUILD_CONFIGURATION}}/net6.0-windows/FrontEndTest.dll
 
-  Beack-End-Test:
```
Comment (slok1368) [inline .github/workflows/main.yml:46]: Good catch

File: src/database/dbObjects.h [hunk 3]
```diff
@@ -28,10 +28,17 @@ namespace gamesDB{
...
-                // Copy name, path, and image path.
+               // Copy name, path, and image path.
                 strncpy_s(name_, name.c_str(), NAME_SIZE);
                 strncpy_s(path_, path.c_str(), PATH_SIZE);
                 strncpy_s(image_path_, image_path.c_str(), PATH_SIZE);
+                // strncpy(name_, name.c_str(), NAME_SIZE);
+                //                 name_[NAME_SIZE - 1] = '\0';
+                //                 strncpy(path_, path.c_str(), PATH_SIZE);
+                //                 path_[PATH_SIZE - 1] = '\0';
+                //                 strncpy(image_path_, image_path.c_str(), PATH_SIZE);
```
Comment (slok1368) [inline src/database/dbObjects.h]: Not important: should we keep these comments

File: src/backend/CMakeLists.txt [hunk 4]
```diff
@@ -17,19 +17,19 @@ FetchContent_MakeAvailable(googletest)
 enable_testing()
 
 add_executable(
-  hello_test
-  hello_test.cc
+  backend_testing
```
Comment (asagrekin) [inline src/backend/CMakeLists.txt:20]: I like the new naming! Let's strive for more intuitive names like this!

File: src/backend/gamelauncher.h [hunk 5]
```diff
@@ -33,30 +35,37 @@ namespace launcher {
...
+    // Produces an updated list of dbObjects by displaying what is in the database.
     void gameList();
 
-    // return the current list of gamesDB::dbObject stored in the database
-    list<gamesDB::dbObject*>* get_data();
 
+    // Deletes a game stored in the database with the given id value.
+    // Input:  An integer value that represents the desired id of the game to delete.
+    // Output: A string to represent the status of the delete.
+    string del(int id);
```
Comment (lochoa2) [inline src/backend/gamelauncher.h:63]: good comments

File: src/backend/gamelauncher.cpp (hunk 1 above)
Comment (lochoa2) [inline src/backend/gamelauncher.cpp]: you might want to delete unused code

File: src/backend/backend_testing.cc [hunk 6]
```diff
@@ -0,0 +1,518 @@
...
+#include "../database/dbObjects.h"  // DataBase Object
+#include <gtest/gtest.h>            // Google Testing Suite
+
+using namespace std;
+
+
+
+//                             NOTES FOR TESTING
+// This application heavily relies on file paths from one's local computer. These tests
+// are catered to a windows device and are using pre-installed file paths such as
```
Comment (slok1368) [inline src/backend/backend_testing.cc:12]: Very detailed, good job

**Review by slok1368 — APPROVED** [review]
Good job on adding the new tests.

File: src/backend/gamelauncher.h [hunk 7]
```diff
@@ -33,30 +35,37 @@ namespace launcher {
...
+    // Plays a game stored in the database with the given id value.
+    // Input:  An integer value that represents the desired id of the game to play.
+    // Output: A boolean value that returns true if play was successful, otherwise false.
     BOOL play(int id);
 
-    // used for deleting a game
-    string del(int id);
 
-    // used for getting updated list of dbObjects
+    // Produces an updated list of dbObjects by displaying what is in the database.
```
Comment (asagrekin) [inline src/backend/gamelauncher.h:56]: Leave implementation details for developers, no need to write them in the header doc.

**Review by lochoa2 — COMMENTED** [review]
review done

**Review by asagrekin — CHANGES_REQUESTED** [review]
Overall, looks very good. Could use some edits based on the comments Leo and Steven left.

**Review by asagrekin — APPROVED** [review]

**Comment by asagrekin** [comment]
LGTM!

**Comment by Kelbaso** [comment]
All reviewer suggestions are resolved ready to merge!

**Pull request merged**